packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -47,18 +47,21 @@ export class WorkspaceFrontendContribution implements CommandContribution, MenuC
     }
 
     protected showFileDialog(): void {
-        this.workspaceService.root.then(root => {
-            const rootUri = new URI(root.uri).parent;
-            this.fileSystem.getFileStat(rootUri.toString()).then(startWith => {
-                const node = DirNode.createRoot(startWith);
+        this.workspaceService.rootResolved.then(resolved => {
+            const root = resolved ? this.workspaceService.root : this.workspaceService.defaultRoot;
+            root.then(root => {
```

Comment:
thought the same, it would be nice to have 2 methods only like `this.root` and `this.tryRoot`, the former is resolved only when the root is set, the latter fallbacks to the default if it is not set

Reply:
Is there a situation where we would not want the default to be used?

Reply:
for example for the navigator, I think it should not show up till the real root is set, i.e. some project is opened

Reply:
> Is there a situation where we would not want the default to be used?

Yes, everywhere. If you do not have a workspace root, then you cannot do anything, right?

The default is needed to feed the workspace root selector dialog.

Reply:
I guess I'm confused from the PR comment : `Workspace root could be unset (undefined) so downstream services
should handle it gracefully.`

So I thought the idea is that you would not need a default ?

Reply:
Should not the root selector be independent of that ? and use it's own default like the home directory?

Reply:
I just read @akosyakov comment about tryRoot and I think I get the idea now... so you can disregard my last comments.